### PR TITLE
feat: Added express-session middleware for expressots

### PR DESCRIPTION
**Pull request by hayawata3626**
# Pull Request Guidelines

Our guidelines for submitting a pull request.

## Before submitting a Pull Request, please make sure you have verified the following:

- [x] The commit message follows our guidelines:
  - A good commit message should be two things: meaningful and concise. It should not contain every single detail, describing each changed line—we can see all the changes in Git—but, at the same time, it should say enough to avoid ambiguity.
  - We use Microverse's commit message convention
  - The convention stablish that a commit message has to be in the present tense, imperative and lowercase.
  - Example: `fix typo in README.md`
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Please describe the current behavior that you are modifying, or link to a relevant issue.

Issue Number: https://github.com/expressots/expressots/issues/124

## What is the new behavior?

Describe the new behavior or link to a relevant issue.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications below.

## Other information

Any other information that is important to this PR.

### CODE REVIEW

**Comment by hayawata3626** [comment]
@rsaz 
I added the code, but I am not quite sure how to test it. Please point out any changes that go against the intent at this point.

**Comment by rsaz** [comment]
> @rsaz I added the code, but I am not quite sure how to test it. Please point out any changes that go against the intent at this point.

For you to test your changes please follow the guideline [here](https://github.com/expressots/expressots/pull/120#issuecomment-1751813527)

File: packages/core/src/middleware/middleware-resolver.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ class MiddlewareResolver {
     morgan: "morgan",
     helmet: "helmet",
     rateLimit: "express-rate-limit",
+    session: "exporess-session",
```
Comment (rsaz) [inline packages/core/src/middleware/middleware-resolver.ts:31]: There is a typo here: `session: "exporess-session"`

Correct is: `session: "express-session"`

File: packages/core/src/middleware/middleware-service.ts
Comment (rsaz) [inline packages/core/src/middleware/middleware-service.ts:1]: You're missing to add the `addSession()` in the interface

`interface IMiddleware { }`

**Review by rsaz — CHANGES_REQUESTED** [review]

File: packages/core/src/middleware/middleware-resolver.ts (hunk 1 above)
Comment (hayawata3626) [reply packages/core/src/middleware/middleware-resolver.ts]: @rsaz 
Sorry. I modified.

File: packages/core/src/middleware/middleware-service.ts
Comment (hayawata3626) [reply packages/core/src/middleware/middleware-service.ts:1]: @rsaz 
I added :pray:

**Review by rsaz — APPROVED** [review]

**Comment by rsaz** [comment]
Close #124

**Pull request merged**